mlflow/utils/string_utils.py

```diff
@@ -61,3 +64,68 @@ def _create_table(
         ),
     ]
     return "\n".join(aligned_rows)
+
+
+# Source: https://github.com/smoofra/mslex/blob/3338c347324d52af619ba39cebfdf7cbf46fa51b/mslex.py#L89-L139
+cmd_meta = r"([\"\^\&\|\<\>\(\)\%\!])"
+cmd_meta_or_space = r"[\s\"\^\&\|\<\>\(\)\%\!]"
+cmd_meta_inside_quotes = r"([\"\%\!])"
+
+
+def mslex_quote(s, for_cmd=True):
+    """
+    Quote a string for use as a command line argument in DOS or Windows.
+
+    On windows, before a command line argument becomes a char* in a
+    program's argv, it must be parsed by both cmd.exe, and by
+    CommandLineToArgvW.
+
+    If for_cmd is true, then this will quote the string so it will
+    be parsed correctly by cmd.exe and then by CommandLineToArgvW.
+
+    If for_cmd is false, then this will quote the string so it will
+    be parsed correctly when passed directly to CommandLineToArgvW.
+
+    For some strings there is no way to quote them so they will
+    parse correctly in both situations.
+    """
+    if not s:
+        return '""'
+    if not re.search(cmd_meta_or_space, s):
+        return s
+    if for_cmd and re.search(cmd_meta, s):
+        if not re.search(cmd_meta_inside_quotes, s):
+            m = re.search(r"\\+$", s)
+            if m:
+                return '"' + s + m.group() + '"'
+            else:
+                return '"' + s + '"'
+        if not re.search(r"[\s\"]", s):
+            return re.sub(cmd_meta, r"^\1", s)
+        return re.sub(cmd_meta, r"^\1", mslex_quote(s, for_cmd=False))
+    i = re.finditer(r"(\\*)(\"+)|(\\+)|([^\\\"]+)", s)
+
+    def parts():
+        yield '"'
+        for m in i:
+            _, end = m.span()
+            slashes, quotes, onlyslashes, text = m.groups()
+            if quotes:
+                yield slashes
+                yield slashes
+                yield r"\"" * len(quotes)
+            elif onlyslashes:
+                if end == len(s):
+                    yield onlyslashes
+                    yield onlyslashes
+                else:
+                    yield onlyslashes
+            else:
+                yield text
+        yield '"'
+
+    return "".join(parts())
+
+
+def quote(s):
+    return mslex_quote(s) if os.name == "nt" else shlex.quote(s)
```

Comment:
Looks like we also use `shlex.quote()` elsewhere in `databricks.py`: https://github.com/mlflow/mlflow/blob/f65429478255b88a211d7dd262c159b4939b44b2/mlflow/projects/databricks.py#L12. Should we change this to `quote`?

Reply:
Thanks for the catch! Fixed.